### PR TITLE
test(ICRC_Ledger): FI-1589: Adapt icrc golden state tests to ledger V3 stable structures migration

### PR DESCRIPTION
**Pull request by mbjorkqvist**
The ICRC ledger does not support downgrading once allowances have been migrated to stable structures. This PR adapts the golden state tests accordingly.

### CODE REVIEW

**Review by basvandijk — APPROVED** [review]

File: mainnet-canisters.json [hunk 1]
```diff
@@ -79,6 +79,10 @@
     "rev": "e54d3fa34ded227c885d04e64505fa4b5d564743",
     "sha256": "3d808fa63a3d8ebd4510c0400aa078e99a31afaa0515f0b68778f929ce4b2a46"
   },
+  "sns_ledger_v2": {
```
Comment (aterga) [inline mainnet-canisters.json:82]: So does `sns_ledger_v2` point to ledger V3? :-)

**Comment by aterga** [comment]
Should we also adopt the tests from https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/nervous_system/integration_tests/tests/sns_ledger_upgrade.rs?L29
?

**Review by aterga — APPROVED** [review]

File: mainnet-canisters.json (hunk 1 above)
Comment (mbjorkqvist) [reply mainnet-canisters.json:82]: This points to V2 of the ledger wrt. the migration to stable structures as define in [the design doc](https://docs.google.com/document/d/1v34PJ0sgYKIKKF2r4o_qhVaeSHqeUwhmR7XnuE5UB0o/edit?tab=t.0#heading=h.mhhrcapko22j), which supports reading both "raw" stable memory, as well as stable memory written using the `MemoryManager`, in the `post_upgrade`. In the `pre_upgrade`, the serialized ledger state is written using the `MemoryManager`.

**Comment by mbjorkqvist** [comment]
> Should we also adopt the tests from https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/nervous_system/integration_tests/tests/sns_ledger_upgrade.rs?L29 ?

From a quick glance, neither of those test would seem to need updating, since they both first install the SNS using the mainnet canister versions (which is the latest blessed versions, i.e., for the ledger, "V2", which writes the ledger state using the `MemoryManager` in the `pre_upgrade`).

The problem is with (SNS) ledgers that haven't upgraded to V2, since we don't support upgrading from a pre-V2 version to V3 or later (to migrate the Allowances in V3 to a virtual memory managed by the `MemoryManager`, the ledger state must have been written using the `MemoryManager` in the `pre_upgrade` of the previous ledger version, and this is only done from V2 onwards).

Since SNSs can (currently?) only upgrade one version at a time, and not skip versions, there shouldn't be any issues, but let's perhaps sync offline to see if there are any potential gaps that we haven't considered.

**Review by maciejdfinity — APPROVED** [review]

**Pull request merged**